app/src/main/java/to/dev/dev_android/view/main/view/CustomWebViewClient.kt

```diff
@@ -39,22 +42,13 @@ class CustomWebViewClient(private val context: Context, private val binding: Act
             }
         }
 
-        for (i in 0 until overrideUrlList.size) {
-            if (url.contains(overrideUrlList[i])) {
-                return false
-            }
-        }
+        if (overrideUrlList.any { url.contains(it) }) return false
```

Comment:
Enclose the `return` statement within code-block. Using one-liners like this are not good for readability and can sometimes lead to big mistakes.

Make it like this:
```kotlin
if (overrideUrlList.any { url.contains(it) }) {
  return false
}
```

And an even better way could be like this:
```kotlin
if (overrideUrlList.any { url.contains(it) }) {
  return false
} else {
  // The rest of your function goes here
}
```

Reply:
I agree on having most of the time a single `{}` for every conditional.

can you give me examples of big mistakes that could happen for the omission of braces like this?

-----

On the other hand, if you want to add the `else` branch it would be better to have:

```kotlin
return if (overrideUrlList.any { url.contains(it) }) {
    false
} else {
    CustomTabsIntent.Builder()
        .setToolbarColor(Color.parseColor("#00000000"))
        .build()
        .also { it.launchUrl(context, Uri.parse(url)) }
    
    true
}
```  

I personally don't think this is needed as the code in the `else` branch will take more space and new identation than a single return and it's a common practice to have single `if(…) { return … }` blocks (with appropriated new lines) but if you think this applies as a better practice for everyone we can add it.

Reply:
Visit [this link](https://nakedsecurity.sophos.com/2014/02/24/anatomy-of-a-goto-fail-apples-ssl-bug-explained-plus-an-unofficial-patch/) for example of a serious mistake that was caused by not using `{}` with single-statement conditionals. **Note:** Search for `/* Mistake` and it will take you straight to the code-snippet followed by explanation.

---

It is true that it would be better to have it the `return` outside of the `if-else` block if we put an else branch.

Your point regarding the additional indentation and it being a common practice to have single `if(...) { return ... }` blacks.

Rather than going with the extracted `return` in an if-else block. I suggest to keep the currently written format like it is except to just add the `{}` around `return false`. So finally it would look like this
```kotlin
if (overrideUrlList.any { url.contains(it) }) {
  return false
}

// The rest of your function goes here
```

Reply:
The example you shared is a good example but also I don't think it's the best, it looks like really bad code anyway.

Anyway, my last commit added the changes we discussed.